src/test/java/hudson/plugins/ec2/EC2CloudMigrationTest.java

```diff
@@ -0,0 +1,39 @@
+package hudson.plugins.ec2;
+
+import com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey;
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
+import jenkins.model.Jenkins;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+public class EC2CloudMigrationTest {
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    @Test
+    @LocalData
```

Comment:
Could you add some comment explaining that the local data is from a previous version of EC2 to show it clear the migration works? Or maybe in the config.xml you can add a comment noticing the `privateKey` is the actual key and not a credential and it works.

Reply:
Added.